pkg/diagnose/connectivity/core_endpoint.go

```diff
@@ -213,3 +227,25 @@ func noResponseHints(err error) string {
 
 	return ""
 }
+
+func clientWithOneRedirects() *http.Client {
+	clientCustom := &http.Client{}
+	clientCustom.CheckRedirect = func(req *http.Request, via []*http.Request) error {
+		return http.ErrUseLastResponse
+	}
+	return clientCustom
+}
+
+// See if the URL is redirected to another URL, and return the status code of the redirection
+func acceptRedirection(url string, client *http.Client) (int, error) {
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏Would be nice having some unit tests for this function as the implementation is not trivial

Reply:
Yes I can do it, wasn't sure if i needed to open an other pr for that or if i do it directly in this one